raiden/tests/utils/mediation_fees.py

```diff
@@ -118,11 +78,75 @@ def amount_without_fees(self) -> PaymentAmount:
         return PaymentAmount(self.total_amount - sum(self.mediation_fees))
 
 
-def get_initial_payment_for_final_target_amount(
-    final_amount: PaymentAmount, channels: List[NettingChannelState]
+def get_amount_before_fees(
+    final_amount: PaymentWithFeeAmount,
+    payer_balance: Balance,
+    payee_balance: Balance,
+    payer_fee_schedule: FeeScheduleState,
+    payee_fee_schedule: FeeScheduleState,
+    payer_capacity: TokenAmount,
+) -> Optional[PaymentWithFeeAmount]:
+    """ Return the amount the transfer requires before fees are deducted.
+
+    This function is also used by the PFS. Therefore the parameteres should not be Raiden state
+    objects.
+    """
+    assert (
+        payer_fee_schedule.cap_fees == payee_fee_schedule.cap_fees
+    ), "Both channels must have the same cap_fees setting for the same mediator."
+    try:
+        fee_func = FeeScheduleState.mediation_fee_backwards_func(
+            schedule_in=payer_fee_schedule,
+            schedule_out=payee_fee_schedule,
+            balance_in=payer_balance,
+            balance_out=payee_balance,
+            capacity_in=payer_capacity,
+            amount_without_fees=final_amount,
+            cap_fees=payer_fee_schedule.cap_fees,
+        )
+    except UndefinedMediationFee:
+        return None
+
+    def angle_bisector(i: int) -> float:
+        x = fee_func.x_list[i]
+        return x - final_amount
+
+    i = 0  # len(fee_func.x_list) - 1
+    y = fee_func.y_list[i]
+    if y < angle_bisector(i):
+        # TODO: can this happen? Should we throw an exception?
+        return None
+    while y >= angle_bisector(i):
```

Comment:
According to one PFS test we should have only a `>` here.

Reply:
I only merged the mediation_fee_funcs so far. Using `>` and `>=` should be mostly the same (depending on how it's handled below). In either case the solution is on one of the joints of the piecewise function, it just changes whether it's the left or right end of the section.